modules/openapi-generator-maven-plugin/src/main/java/org/openapitools/codegen/plugin/CodeGenMojo.java

```diff
@@ -361,7 +372,17 @@ public void execute() throws MojoExecutionException {
             configurator.setIgnoreFileOverride(ignoreFileOverride);
         }
 
-        configurator.setLang(language);
+        // TODO: After 3.0.0 release (maybe for 3.1.0): Fully deprecate lang.
+        if (isNotEmpty(generatorName)) {
+            configurator.setGeneratorName(generatorName);
+        } else if (isNotEmpty(language)) {
+            LOGGER.warn("The 'language' option is deprecated and may reference language names only in the next major release (4.0). Please use 'generatorName' instead.");
+            configurator.setGeneratorName(language);
+        } else {
+            LOGGER.error("A generator name (generatorName) is required.");
+            System.exit(1);
```

Comment:
I think it is better to throw a `MojoExecutionException` instead of System exist (maven will terminate with an appropriate error message and can still do some other stuff)

Reply:
Thanks for catching this. I was just blindly copying what I added to the CLI code, and I can't believe I left System.exit here. 💃